### PR TITLE
Fc bls walletlist RPC

### PR DESCRIPTION
**Pull request by jframe**
Filecoin WalletList RCP implementation for BLS

### CODE REVIEW

File: core/src/main/java/tech/pegasys/eth2signer/core/service/jsonrpc/SigningService.java [hunk 1]
```diff
@@ -89,6 +91,8 @@ public String upcheck() {
 
   @JsonRpcMethod("Filecoin.WalletList")
   public List<String> filecoinWalletList() {
-    return fcKeyIdentifiers.list(KeyType.SECP256K1);
+    final List<String> secp = fcKeyIdentifiers.list(KeyType.SECP256K1);
+    final List<String> bls = fcKeyIdentifiers.list(KeyType.BLS);
+    return Stream.concat(secp.stream(), bls.stream()).collect(Collectors.toList());
```
Comment (usmansaleem) [inline core/src/main/java/tech/pegasys/eth2signer/core/service/jsonrpc/SigningService.java:96]: nit: list = new ArrayList<>(secp); list.addAll(bls) might be faster. but its just a nit pick.

**Review by usmansaleem — APPROVED** [review]
LGTM

File: core/src/main/java/tech/pegasys/eth2signer/core/service/jsonrpc/SigningService.java (hunk 1 above)
Comment (jframe) [reply core/src/main/java/tech/pegasys/eth2signer/core/service/jsonrpc/SigningService.java:96]: done

**Review by rain-on — APPROVED** [review]
LGTM

File: core/src/main/java/tech/pegasys/eth2signer/core/multikey/metadata/BlsArtifactSignerFactory.java [hunk 2]
```diff
@@ -35,24 +35,27 @@
...
 
   private final LabelledMetric<OperationTimer> privateKeyRetrievalTimer;
+  private final Function<BLSKeyPair, ArtifactSigner> signerFactory;
 
   public BlsArtifactSignerFactory(
       final Path configsDirectory,
       final MetricsSystem metricsSystem,
-      final HashicorpConnectionFactory connectionFactory) {
+      final HashicorpConnectionFactory connectionFactory,
+      final Function<BLSKeyPair, ArtifactSigner> signerFactory) {
```
[the remark this answers is not in the document]
Comment (jframe) [reply core/src/main/java/tech/pegasys/eth2signer/core/multikey/metadata/BlsArtifactSignerFactory.java:44]: yes same approach

**Pull request merged**